pkg/asset/ignition/machine/azure/node.go

```diff
@@ -0,0 +1,17 @@
+package azure
```

Comment:
this needs to be through machinconfig like [here](https://github.com/openshift/installer/tree/master/pkg/asset/machines/machineconfig) and [here](https://github.com/openshift/installer/blob/master/pkg/asset/machines/master.go#L154) the pointer configs serve the purpose of **only** providing hosts where to fetch the igntionconfig ie from inside the cluster.

Reply:
this change is needed for kubelet to register the right host name with kubernetes, cannot use machine config operator, since that operation is performed after kubelet starts. So the only place would be part of ignition config

Reply:
no, we run machine-config-operator on bootstrap node to provide configuration for control plane hosts, so it will be correctly setup.

Reply:
and bootstrap host kubelet runs only in local mode so bootstrap ignition doesn't need this change.

Reply:
Thanks @abhinavdahiya makes sense now, we were always under assumption that MCO is used only for runtime configuration. We will move provider specific implementation to MCO instead.

There are so many ways of doing configuration since we arent aware of the history and architecture background could be mistaken for shortcuts, please do bear with us while we are learning along.

@serbrech I can take care of moving this to MCO instead.

Reply:
Just note that we are talking about a temporary solution, which is still a workaround.
The proper place for this is in Afterburn. so it's still misplaces, whether it's in MCO or in ignition.

Reply:
@abhinavdahiya seems like there is huge backlash on any new functionality changes added to MCO, https://github.com/openshift/machine-config-operator/pull/602#issuecomment-481797061 , considering that, we want to progress with this PR instead as @serbrech mentioned this is temporary work around and will be removed in the future once afterburn changes are integrated, Will that work for now?

Reply:
the change i requested https://github.com/openshift/installer/pull/1570#discussion_r273752111 is not changing on MCO.. but in the installer... So i'm confused.